File: Server/routes/CompanyRoutes.js
const express = require('express');
const router = express.Router();
const companyController = require('../controllers/companyController.js');
const isAuthenticated = require('../Utils/AuthenticationUtil.js');

router.get('/get', isAuthenticated, companyController.getCompany);

router.get('/all', isAuthenticated, companyController.getCompanies);

router.post('/update', isAuthenticated, companyController.updateCompany);

router.post('/create', isAuthenticated, companyController.createCompany);

module.exports = router;
 

File: Server_Go/routes/companyRoutes.go
package routes

import (
	"github.com/gorilla/mux"
)

func RegisterCompanyRoutes(r *mux.Router) {
	r.HandleFunc("/get", companyController.getCompany).Name("Get Company")
	r.HandleFunc("/all", companyController.getCompanies).Name("Get Companies")
	r.HandleFunc("/update", companyController.updateCompany).Name("Update Company")
	r.HandleFunc("/create", companyController.createCompany).Name("Create Company")
}
